selfdrive/loggerd/loggerd.cc

```diff
@@ -298,89 +217,92 @@ void encoder_thread(RotateState *rotate_state, bool is_streaming, bool raw_clips
         break;
       }
 
-      //uint64_t current_time = nanos_since_boot();
-      //uint64_t diff = current_time - extra.timestamp_eof;
-      //double msdiff = (double) diff / 1000000.0;
-      // printf("logger latency to tsEof: %f\n", msdiff);
-
-      { // all the rotation stuff
-
-        pthread_mutex_lock(&s.rotate_lock);
-        pthread_mutex_unlock(&s.rotate_lock);
-
-        // wait if camera pkt id is older than stream
-        rotate_state->waitLogThread();
-
-        if (do_exit) break;
+      // Rules of rotation:
+      // 1.rotate all cameras on first time run.
+      // 2.rotate all cameras after main thread's time-based rotation occur.
+      // 3.rotate front camera after main thread rotated.
+      // 4.rotate rear&wide camera after (segment_length*MAIN_FPS) frames have been encoded.
+
+      bool should_rotate = encoder_segment != s.rotate_segment;  // true if first time run or time-based rotation occur
+      if (!should_rotate && cam_idx != LOG_CAMERA_ID_DCAMERA) {  // rear&wide camera
```

Comment:
on tici we still want DCAM to rotate based off its own frame ids

Reply:
fixed